lib/std/build.zig

```diff
@@ -2542,89 +2561,93 @@ pub const LibExeObjStep = struct {
             try zig_args.append(builder.pathFromRoot(version_script));
         }
 
-        if (self.exec_cmd_args) |exec_cmd_args| {
-            for (exec_cmd_args) |cmd_arg| {
-                if (cmd_arg) |arg| {
-                    try zig_args.append("--test-cmd");
-                    try zig_args.append(arg);
-                } else {
-                    try zig_args.append("--test-cmd-bin");
-                }
-            }
-        } else {
-            const need_cross_glibc = self.target.isGnuLibC() and self.is_linking_libc;
-
-            switch (self.builder.host.getExternalExecutor(self.target_info, .{
-                .qemu_fixes_dl = need_cross_glibc and builder.glibc_runtimes_dir != null,
-                .link_libc = self.is_linking_libc,
-            })) {
-                .native => {},
-                .bad_dl, .bad_os_or_cpu => {
-                    try zig_args.append("--test-no-exec");
-                },
-                .rosetta => if (builder.enable_rosetta) {
-                    try zig_args.append("--test-cmd-bin");
-                } else {
-                    try zig_args.append("--test-no-exec");
-                },
-                .qemu => |bin_name| ok: {
-                    if (builder.enable_qemu) qemu: {
-                        const glibc_dir_arg = if (need_cross_glibc)
-                            builder.glibc_runtimes_dir orelse break :qemu
-                        else
-                            null;
+        if (self.kind == .@"test") {
+            if (self.exec_cmd_args) |exec_cmd_args| {
+                for (exec_cmd_args) |cmd_arg| {
+                    if (cmd_arg) |arg| {
```

Comment:
This got mangled in the diff but it's just an indentation change, under the condition `self.kind == .@"test"`.  There's also an added else branch at the bottom.